spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/StoredProcOutboundGateway.java

```diff
@@ -56,6 +87,14 @@ public String getComponentType() {
 	}
 
 	@Override
+	protected void doInit() {
+		super.doInit();
+		if (!this.requiresReplyExplicitlySet) {
+			setRequiresReply(this.expectSingleResult);
+		}
```

Comment:
I was suggesting keeping the flag and using it in the setter; that way, it doesn't matter which order the setters are called. But, maybe that's too much logic in a setter and, in any case, this component won't work properly if it's configured outside of Spring; so I am ok with this. Will merge soon.